### PR TITLE
Maintenance : simplify a few patterns, remove unneeded dependencies

### PR DESCRIPTION
**Pull request by huitseeker**
Opinions on the refactorings loosely held -> happy to accomodate in review

#### Summary of Changes

Fixes #

### CODE REVIEW

File: bench-exchange/Cargo.toml [hunk 1]
```diff
@@ -9,19 +9,14 @@ homepage = "https://solana.com/"
...
 itertools = "0.8.2"
 log = "0.4.8"
 num-derive = "0.3"
 num-traits = "0.2"
 rand = "0.6.5"
 rayon = "1.2.0"
-serde = "1.0.104"
-serde_derive = "1.0.103"
-serde_json = "1.0.46"
+serde_json = "1.0.44"
```
Comment (garious) [inline bench-exchange/Cargo.toml]: Needs a rebase

File: crate-features/Cargo.toml [hunk 2]
```diff
@@ -9,20 +9,7 @@ license = "Apache-2.0"
 edition = "2018"
 
 [dependencies]
```
Comment (garious) [inline crate-features/Cargo.toml]: This crate is special. Check out its README. If still not clear, ping @mvines.

File: runtime/src/accounts_index.rs [hunk 3]
```diff
@@ -75,11 +75,8 @@ impl<T: Clone> AccountsIndex<T> {
     ) -> Option<(RwLockReadGuard<SlotList<T>>, usize)> {
         self.account_maps.get(pubkey).and_then(|list| {
             let lock = list.read().unwrap();
-            if let Some(found_index) = self.latest_slot(ancestors, &lock) {
-                Some((lock, found_index))
-            } else {
-                None
-            }
+            self.latest_slot(ancestors, &lock)
```
Comment (garious) [inline runtime/src/accounts_index.rs]: How about:

```rust
let found_index = self.latest_slot(ancestors, &lock)?;
(lock, found_index)
```

Comment (huitseeker) [reply runtime/src/accounts_index.rs]: Thanks!  This doesn't work because it would return the tuple rather than an `Option`.

Comment (garious) [reply runtime/src/accounts_index.rs]: Right, would need to be `Some((lock, found_index))`

File: programs/config/src/config_processor.rs [hunk 4]
```diff
@@ -96,7 +96,7 @@ pub fn process_instruction(
         return Err(InstructionError::InvalidInstructionData);
     }
 
-    config_keyed_account.try_account_ref_mut()?.data[0..data.len()].copy_from_slice(&data);
+    config_keyed_account.try_account_ref_mut()?.data[..].copy_from_slice(&data);
```
Comment (garious) [inline programs/config/src/config_processor.rs]: Looks like this change broke a bunch of tests.

**Comment by garious** [comment]
Can you run the build, such that the updated Cargo.lock is included in this PR?

File: programs/config/src/config_processor.rs (hunk 4 above)
Comment (huitseeker) [reply programs/config/src/config_processor.rs]: I have trouble figuring out [why](https://github.com/rust-lang/rust/blob/master/src/libcore/ops/range.rs#L26-L36). I suspect a dependency interaction.

Comment (garious) [reply programs/config/src/config_processor.rs]: `data != config_keyed_account.try_account_ref_mut()?.data`

Comment (garious) [reply programs/config/src/config_processor.rs]: Deleting just the zero from the original code would obfuscate things a little less.

Comment (huitseeker) [reply programs/config/src/config_processor.rs]: :man_facepalming:

**Comment by garious** [comment]
Got a tool that spotted all the redundant crate dependencies?

**Comment by huitseeker** [comment]
@garious Nope. I literally brute-forced them (remove one, try to see if it still works ..).

**Comment by garious** [comment]
Thanks so much @huitseeker! A wonderful contribution!

**Comment by jackcmay** [comment]
Lots of great cleanup, thanks!

**Comment by ryoqun** [comment]
> @garious Nope. I literally brute-forced them (remove one, try to see if it still works ..).

@huitseeker Nice work! I'm pretty amazed by such a great hard manual work. :)

**Comment by huitseeker** [comment]
@ryoqun oh, I scripted the trial-and error, but that's just a couple lines of bash, and what you'd come up with is as good as what I did. Let me know if you want me to post this as a gist.

**Comment by ryoqun** [comment]
@huitseeker Yeah, I'd rather want to view the script out of my pure curiosity. Because I like these cleaning-ups a lot. So that is for my future work inspiration :)

**Comment by huitseeker** [comment]
@ryoqun It's a slight variant of what's in https://github.com/libra/libra/pull/2528

**Comment by mvines** [comment]
I wish there was a clippy lint for unused dependencies.  :-/

**Comment by huitseeker** [comment]
@mvines there is https://crates.io/crates/cargo-udeps but I've found its workspace and feature support lacking

**Comment by ryoqun** [comment]
@huitseeker thanks for the pointer!

**Comment by ryoqun** [comment]
> I wish there was a clippy lint for unused dependencies. :-/

> @mvines there is https://crates.io/crates/cargo-udeps but I've found its workspace and feature support lacking

dreams comes true: #31320

**Pull request merged**